x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/enrollment/CreateEnrollmentTokenTests.java

```diff
@@ -0,0 +1,238 @@
+            final URL getHttpInfoURL = CreateEnrollmentToken.getHttpInfoURL(defaultUrl);
+
+            final HttpResponse httpResponseOK = new HttpResponse(HttpURLConnection.HTTP_OK, new HashMap<String, Object>());
+            when(client.execute(anyString(), any(URL.class), anyString(), any(SecureString.class), any(CheckedSupplier.class),
+                any(CheckedFunction.class))).thenReturn(httpResponseOK);
+
+            String createApiKeyResponseBody;
+            try (XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON)) {
+                builder.startObject()
+                    .field("id", "DR6CzXkBDf8amV_48yYX")
+                    .field("name", "enrollment_token_API_key_VuaCfGcBCdbkQm")
+                    .field("expiration", "1622652381786")
+                    .field("api_key", "x3YqU_rqQwm-ESrkExcnOg")
+                    .endObject();
+                createApiKeyResponseBody = Strings.toString(builder);
+            }
+            when(client.execute(eq("POST"), eq(createAPIKeyURL), eq(ElasticUser.NAME), any(SecureString.class),
+                any(CheckedSupplier.class), any(CheckedFunction.class)))
+                .thenReturn(createHttpResponse(HttpURLConnection.HTTP_OK, createApiKeyResponseBody));
+
+            String getHttpInfoResponseBody;
+            try (XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON)) {
+                builder.startObject()
+                    .startObject("nodes")
+                    .startObject("sxLDrFu8SnKepObrEOjPZQ")
+                    .field("version", "8.0.0")
+                    .startObject("http")
+                    .field("publish_address", "127.0.0.1:9200")
+                    .endObject().endObject().endObject().endObject();
+                getHttpInfoResponseBody = Strings.toString(builder);
+            }
+            when(client.execute(eq("GET"), eq(getHttpInfoURL), eq(ElasticUser.NAME), any(SecureString.class),
+                any(CheckedSupplier.class), any(CheckedFunction.class)))
+                .thenReturn(createHttpResponse(HttpURLConnection.HTTP_OK, getHttpInfoResponseBody));
+
+            final CreateEnrollmentToken createEnrollmentToken = new CreateEnrollmentToken(environment, client);
+
+            final String response = createEnrollmentToken.create(terminal, "elastic", new SecureString("elastic"));
+
+            final String version = response.substring(0, response.lastIndexOf("."));
+            final String token = response.substring(response.lastIndexOf(".") + 1);
+            assertEquals(version, "8.0.0");
+            Map<String, String> info = getDecoded(token);
+            assertEquals("[127.0.0.1:9200]", info.get("adr"));
+            assertEquals("598a35cd831ee6bb90e79aa80d6b073cda88b41d", info.get("fgr"));
+            assertEquals("x3YqU_rqQwm-ESrkExcnOg", info.get("key"));
+        } catch (Exception e) {
```

Comment:
can't we simply declare the test method throwing an Exception ? Same in all tests below